.github/workflows/test.yml

```
@@ -0,0 +1,453 @@
+          rails:
+          i18n:
+          gemfile: gemfiles/driver_min.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails: '5.1'
+          i18n:
+          gemfile: gemfiles/rails-5.1.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '5.1'
+          i18n:
+          gemfile: gemfiles/rails-5.1.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails: '5.2'
+          i18n:
+          gemfile: gemfiles/rails-5.2.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '5.2'
+          i18n:
+          gemfile: gemfiles/rails-5.2.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails: '6.0'
+          i18n:
+          gemfile: gemfiles/rails-6.0.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '6.0'
+          i18n:
+          gemfile: gemfiles/rails-6.0.gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-3.0
+          topology:
+          task: test
+          driver: current
+          rails: '6.0'
+          i18n:
+          gemfile: gemfiles/rails-6.0.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.5
+          topology:
+          task: test
+          driver: current
+          rails:
+          i18n: '1.0'
+          gemfile: gemfiles/i18n-1.0.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails:
+          i18n: '1.0'
+          gemfile: gemfiles/i18n-1.0.gemfile
+          experimental: false
+        - mongodb: '4.2'
+          ruby: ruby-2.6
+          topology:
+          task: test
+          driver: current
+          rails:
+          i18n: current
+          gemfile: Gemfile
+          experimental: false
+        - mongodb: '4.4'
+          ruby: ruby-2.7
+          topology:
+          task: test
+          driver: current
+          rails: '5.2'
+          i18n:
+          gemfile: gemfiles/rails-5.2.gemfile
+          experimental: false
+    steps:
+    - name: repo checkout
+      uses: actions/checkout@v2
+    - name: start mongodb
+      uses: supercharge/mongodb-github-action@1.6.0
+      with:
+        mongodb-version: "${{matrix.mongodb}}"
+        mongodb-replica-set: rs0
```

Depending on the configurations being used, we can leverage drivers-evergreen-tools for this:
```suggestion
    - name: start mongodb
      id: "setup-mongodb"
      uses: mongodb-labs/drivers-evergreen-tools@master
      with:
        version: "${{matrix.mongodb}}"
        topology: "${{matrix.topology}}"
```

I'll note that this may require changing the values of `matrix.topology`, so I'll let @p-mongo and @comandeo decide how to proceed here. I'm fine using something other than drivers-evergreen-tools if it makes the job at hand easier.